api/src/main/java/org/eclipse/microprofile/concurrent/spi/ActiveThreadContext.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) 2018 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * You may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.eclipse.microprofile.concurrent.spi;
+
+/**
+ * <p>Represents context that is applied to a particular thread, along with any
+ * state that is associated with it or that is necessary for restoring the
+ * previous context afterward.</p>
+ *
+ * <p>When the context is no longer needed on the thread, the
+ * <code>ManagedExecutor</code> or <code>ThreadContext</code> must invoke the
+ * <code>end</code> method.</p>
+ */
+public interface ActiveThreadContext {
```

Comment:
I would say this is a controller. How about ThreadContextController?

Reply:
I like how the name ThreadContextController starts with ThreadContext in the same way that ThreadContextProvider and ThreadContextSnapshot do, which will cause them to naturally be listed together when interfaces are shown alphabetically.  ActiveThreadContext seems more descriptive/clear, except could be confusing after it is ended, so I'm willing to change it as you suggest.